core/loader_shared.c

```diff
@@ -178,6 +178,7 @@ loader_init_prologue(void)
         privmod_t *mod =
             privload_insert(NULL, privmod_static[i].base, privmod_static[i].size,
                             privmod_static[i].name, privmod_static[i].path);
+        mod->is_top_level_client = true;
```

Comment:
Please update the PR description to make it explicit that the skipping only applies to top-level clients and not extensions/libraries via a new module flag.